### ISSUE TITLE
Doesn't work with Azure Cosmos MongoDB API

### ISSUE THREAD

**Issue by tiagoefreitas** [formio/formio#1092]
## Environment
- Hosting type
  - [x ] Local deployment
    - Version: 1.90.12

## Steps to Reproduce
Configure Azure Cosmos Db, open login page.

## Observed behavior

Get an error in the response:
The index path corresponding to the specified order-by item is excluded

Azure Cosmos doesn't support sorting on a field without an index, for performance reasons, but default MongoDB works.

This solved it by creating the missing index:

db.actions.createIndex( { "$**" : 1 } )

I didn't check which field was missing an index, since above command creates indexes on all fields.

**Comment by wag110894** [comment]
Hello @tiagoefreitas 

Thank you for reaching out! We have created a ticket and have added this to our backlog. We welcome a put request and have a process to expedite if you are interested.

For reference, the ticket number is FOR-2756.

We will let you know once this issue is resolved.

Have a great day!

**Comment by phucduong86** [comment]
@wag110894 
We are also in the process if migrating to CosmosDB and getting the same issue.
I believe we've set it up previously on CosmosDB Mongo API version 3.2 and it works fine.
Can you please confirm if it still works on 3.2 and this issue is with new Cosmos version 3.6?

Thank you

**Comment by randallknutson** [comment]
We researched this issue last week to figure out what was going on and the issue is that in CosmosDB with Mongo 3.6 you can't sort on any property that doesn't have an index set. This is different than how mongoDB works since you CAN sort on any property regardless of if there is an index or not. 

To fix this, you can add a wildcard index to the following collections:
db.submissions.createIndex( {"$**": 1 } );
db.forms.createIndex( {"$**": 1 } );
db.projects.createIndex( {"$**": 1 } );
db.actions.createIndex( {"$**": 1 } );'
db.roles.createIndex( {"$**": 1 } );

**Comment by phucduong86** [comment]
@randallknutson 
We actually had to run an additional query to see the Actions panel:
```db.actionitems.createIndex( {"$**": 1 } );```

I just discovered another issue today and not sure if it is related to the Cosmo query as well:

## Step to reproduce:
- Pick any form
- Add a Select component
- On Data tab choose Resource
- Click the dropdown under Resource.

Expected: see a list of resources in the list
Observerd: list is blank, and the following request failed when I inspect network: 
![image](https://user-images.githubusercontent.com/21692768/98737378-1c207600-2374-11eb-910c-2754c8a7b92e.png)

**Comment by travist** [comment]
We are in the process of updating our documentation to mention this, but for those who need this, here is the commands to run to make this work with Form.io.

```
db.submissions.createIndex( {"$**": 1 } );
db.forms.createIndex( {"$**": 1 } );
db.projects.createIndex( {"$**": 1 } );
db.actions.createIndex( {"$**": 1 } );'
db.roles.createIndex( {"$**": 1 } );
```

**Comment by phucduong86** [comment]
@travist 
We have already ran those commands, we actually had to run an additional one to get the Actions log tab to query:
```
db.actionitems.createIndex( {"$**": 1 } );
```

The Select Component is still an issue after running all 6 commands.
Thank you

**Comment by travist** [comment]
Looks like we are adding a limit that is larger than it supports.  We will add this as a fix in our backlog.

**Comment by wag110894** [comment]
@phucduong86 for reference, the ticket number for this item is FOR-2858.

**Comment by jhen1422** [comment]
This should be resolved in the latest 7.0.0 version of our server
